### PR TITLE
Static Analysis Updates

### PR DESCRIPTION
**Pull request by ejfitzgerald**
- Static analysis can be performed natively under MacOS. (User will need to install `brew install llvm@6`
- Style formatting also now supported natively under MacOS.
- Incorporated @n-hutton diff based style apply (to improve developer workflow)
- Additional MacOS specific formatting updates

### CODE REVIEW

File: scripts/ci-tool.py [hunk 1]
```diff
@@ -327,9 +325,30 @@ def main():
...
+        if isdir(build_root):
+            if isfile(join(build_root, 'Makefile')):
+                generator = 'Unix Makefiles'
+            elif isfile(join(build_root, 'build.ninja')):
+                generator = 'Ninja'
+            else:
+                raise RuntimeError('Unable to detect existing generator type')
+        else:
+
+            # in the case of a new build prefer Ninja over make (because it is faster)
```
Comment (n-hutton) [inline scripts/ci-tool.py:338]: 👍 - this is kindof transparent to the user anyway right?

File: scripts/ci-tool.py [hunk 2]
```diff
@@ -327,9 +325,30 @@ def main():
...
+            generator = 'Ninja' if NINJA_IS_PRESENT else 'Unix Makefiles'
+
+        # due to the version of the header dependency detection that is used when trying to
+        # determine affected files, when running commit filtered lints must use make
+        if generator == 'Ninja' and args.commit is not None:
+            generator = 'Unix Makefiles'
+
+        # configure the project
+        if not cmake_configure(project_root, build_root, options, generator):
+            output('\n😭 Failed to configure the cmake project. This is usually because of a mismatch between generators.\n\nTry removing the build folder: {} and try again'.format(build_root))
```
Comment (n-hutton) [inline scripts/ci-tool.py:348]: a crying emoji? Intended to be here 🤣 ?

File: scripts/fetchai_code_quality/internal/static_analysis.py [hunk 3]
```diff
@@ -51,8 +153,14 @@ def read_static_analysis_yaml(project_root, build_root, yaml_file_path):
         new_diagnostics = []
         for d in document['Diagnostics']:
             file_path = d['FilePath']
+
+            # ignore extries that refer to empty files (usually waning have stopped on a given file)
```
Comment (n-hutton) [inline scripts/fetchai_code_quality/internal/static_analysis.py]: waning typo

File: scripts/fetchai_code_quality/internal/static_analysis.py [hunk 4]
```diff
@@ -127,35 +235,121 @@ def get_disabled_checks_list():
...
+
+            if target_files is not None:
+                match = False
+                for target_file in target_files:
+                    if target_file in command_file_path:
+                        match = True
+                        break
+
+                if not match:
+                    print('Dropping 1', relative_file_path)
```
Comment (n-hutton) [inline scripts/fetchai_code_quality/internal/static_analysis.py]: dropping 2 and dropping 1 intended to be in?

**Review by n-hutton — DISMISSED** [review]

File: scripts/ci-tool.py (hunk 2 above)
Comment (ejfitzgerald) [reply scripts/ci-tool.py:348]: Yes it was intentional

File: scripts/fetchai_code_quality/internal/static_analysis.py (hunk 4 above)
Comment (ejfitzgerald) [reply scripts/fetchai_code_quality/internal/static_analysis.py]: Nope good spot

File: scripts/ci-tool.py (hunk 1 above)
Comment (ejfitzgerald) [reply scripts/ci-tool.py:338]: Correct

**Review by n-hutton — APPROVED** [review]

**Pull request merged**